app_dart/lib/src/model/firestore/task.dart

```diff
@@ -0,0 +1,214 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:googleapis/firestore/v1.dart' hide Status;
+
+import '../../request_handling/exceptions.dart';
+import '../../service/firestore.dart';
+import '../../service/logging.dart';
+import '../luci/push_message.dart';
+
+class Task extends Document {
+  /// Lookup [Task] from Firestore.
+  ///
+  /// `documentName` follows `/projects/{project}/databases/{<}database}/documents/{document_path}`
+  static Future<Task> fromFirestore({
+    required FirestoreService firestoreService,
+    required String documentName,
+  }) async {
+    final Document document = await firestoreService.getDocument(documentName);
+    return Task.fromDocument(taskDocument: document);
+  }
+
+  /// Create [Task] from a task Document.
+  static Task fromDocument({
+    required Document taskDocument,
+  }) {
+    return Task()
+      ..fields = taskDocument.fields!
+      ..name = taskDocument.name!;
+  }
+
+  /// The task was cancelled.
+  static const String statusCancelled = 'Cancelled';
+
+  /// The task is yet to be run.
+  static const String statusNew = 'New';
+
+  /// The task failed to run due to an unexpected issue.
+  static const String statusInfraFailure = 'Infra Failure';
```

Comment:
I feel like I see these enums in a lot of places...

Reply:
We have different statuses in different contexts: the flutter build dashboard task statuses, the buildbucket build statuses, GitHub statuses, etc. 
But here is a direct map from datastore/task statuses, which would be deprecated when firestore logic is ready.

Reply:
Got it!